apps/studio/src/pages/sites/[siteId]/settings/agency.tsx

```diff
@@ -0,0 +1,136 @@
+import FormBuilder from "~/features/editing-experience/components/form-builder/FormBuilder"
+import { UnsavedSettingModal } from "~/features/editing-experience/components/UnsavedSettingModal"
+import { siteSchema } from "~/features/editing-experience/schema"
+import { SettingsEditingLayout } from "~/features/settings/SettingsEditingLayout"
+import { SettingsHeader } from "~/features/settings/SettingsHeader"
+import { useNavigationEffect } from "~/hooks/useNavigationEffect"
+import { useNewSettingsPage } from "~/hooks/useNewSettingsPage"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { useZodForm } from "~/lib/form"
+import { type NextPageWithLayout } from "~/lib/types"
+import { updateSiteConfigSchema } from "~/schemas/site"
+import { SiteSettingsLayout } from "~/templates/layouts/SiteSettingsLayout"
+import { ajv } from "~/utils/ajv"
+import { trpc } from "~/utils/trpc"
+
+const AgencySettingsPage: NextPageWithLayout = () => {
+  const isEnabled = useNewSettingsPage()
+  const router = useRouter()
+  const { siteId: rawSiteId } = useQueryParse(siteSchema)
+  const siteId = Number(rawSiteId)
+  const [{ siteName, agencyName }] = trpc.site.getConfig.useSuspenseQuery({
+    id: siteId,
+  })
+  const trpcUtils = trpc.useUtils()
+  const toast = useToast()
+  const validateFn = ajv.compile<AgencySettings>(AgencySettingsSchema)
+
+  const updateSiteConfigMutation = trpc.site.updateSiteConfig.useMutation({
+    onSuccess: async ({ siteName }) => {
+      toast({
+        title: `Site ${siteName} updated successfully`,
+        status: "success",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+      await trpcUtils.site.getConfig.invalidate({ id: siteId })
+      await trpcUtils.site.getSiteName.invalidate({ siteId })
+    },
+    onError: (error) => {
+      toast({
+        title: "Failed to update site",
+        description: error.message,
+        status: "error",
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+  })
+
+  useEffect(() => {
+    if (!isEnabled) {
+      void router.push(`/sites/${siteId}/settings`)
+    }
+  }, [])
+
+  const { handleSubmit } = useZodForm({
+    schema: updateSiteConfigSchema.omit({ siteId: true }),
+    defaultValues: { siteName },
+  })
+
+  const [nextUrl, setNextUrl] = useState("")
+  const isOpen = !!nextUrl
+  const [state, setState] = useState<AgencySettings>({
+    siteName,
+    agencyName,
+  })
+  const isDirty = state.siteName !== siteName
+
+  useNavigationEffect({ isOpen, isDirty, callback: setNextUrl })
+
+  const onSubmit = handleSubmit(() =>
+    updateSiteConfigMutation.mutate({
+      siteName: state.siteName,
+      siteId,
+    }),
+  )
+
+  return (
+    <>
+      <UnsavedSettingModal
+        isOpen={isOpen}
+        onClose={() => setNextUrl("")}
+        nextUrl={nextUrl}
+      />
+      <chakra.form overflow="auto" height={0} minH="100%" onSubmit={onSubmit}>
+        <SimpleGrid columns={9} h="100%">
+          <SettingsEditingLayout>
```

Comment:
issue: discussed offline, should use the Grid pattern currently on EditPage so that the scrolling only happens within the form editor and does not affect the preview section

Reply:
dsone